test/sqllogictest/temporal.slt

```diff
@@ -0,0 +1,45 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
```

Comment:
This query hangs forever:

```materialize=> DROP TABLE IF EXISTS temporal;
DROP TABLE
materialize=> CREATE TABLE temporal (start_time bigint);
CREATE TABLE
materialize=> select * from temporal where start_time > mz_logical_timestamp() as of to_timestamp(10000000000);
```
I assume some overflow occurs with such large values.

Reply:
I can't tell how big that is (is it in the far future of "now"?) but it is possible this is expected. If you issue an `as of` query with a time that the source has not yet reached, the dataflow layer will enqueue the query and await the source advancing. Our `TABLE` sources use wall clock time, and so if that time happens to be in 2098, your session will be hung until the server thinks it is 2098.

You should still be able to create a new session at the same time, and the system should still be responsive, just this one `select` query will be blocked waiting for the correct answer. Does that seem to line up with your experience here?

Reply:
> Sat Nov 20 2286 12:46:40 UTC-0500 (EST)

You'd have to wait until then. :D

Reply:
Yes, the system remains responsive. I was just hoping that an empty result will be returned immediately, since it was not a tail query or anything of that nature. But I do understand the reasoning now, so we are good.

Reply:
I think we could reasonably discuss changing the behavior to return an error. We wouldn't want to return the empty set because we don't know this to be the correct answer yet, but with TAIL working, there is less of a reason to have SELECT block until a time (if you want that behavior, we might want to point you at TAIL).